tools/nuget/validate_package.py

```diff
@@ -57,13 +68,75 @@ def parse_arguments():
 def check_exists(path):
     return os.path.exists(path)
 
+def remove_residual_files(path):
+    if check_exists(path):
+        os.remove(path)
+
 
 def is_windows():
     return sys.platform.startswith("win")
 
 
+def check_if_headers_are_present(
+    header_files, header_folder, file_list_in_package, platform
+):
+    for header in header_files:
+        path = header_folder + "/" + header
+        print("Checking path: " + path)
+        if path not in file_list_in_package:
+            print(header + " not found for " + platform)
+            raise Exception(header + " not found for " + platform)
+
+def download_and_verify_directml_dlls(zip_file, platform, package_folder):
+    dml_nuget_spec_file_name = "Microsoft.ML.OnnxRuntime.DirectML.nuspec"
+    dml_nuget_package_name = 'Microsoft.AI.DirectML'
+    dml_nuget_package_version = '1.9.1'
+    dml_dependency_string = '<dependency id="' + dml_nuget_package_name + '" version="' + dml_nuget_package_version + '" />'
+
+    dml_related_dlls = [
+        "DirectML.Debug.dll",
+        "DirectML.Debug.pdb",
+        "DirectML.dll",
+        "DirectML.lib",
+        "DirectML.pdb",
+    ]
+
+    with zip_file.open(dml_nuget_spec_file_name, 'r') as file:
+        content = file.read().decode('UTF-8')
+        if dml_dependency_string in content:
+            dml_nupkg_zip = package_folder + '/' + dml_nuget_package_name + '.' + dml_nuget_package_version + '.zip'
+
+            if not check_exists(dml_nupkg_zip):
+                dml_nuget_url = 'https://api.nuget.org/v3-flatcontainer/' + dml_nuget_package_name + '/' + dml_nuget_package_version + '/' + dml_nuget_package_name + '.' + dml_nuget_package_version + '.nupkg'
+                try:
+                    request.urlretrieve(dml_nuget_url, dml_nupkg_zip)
+                except Exception as e:
+                    print(e)
+                    raise Exception("Unable to download" + dml_nuget_url+  "NuGet package")
+
+            dml_zip_file_list = zipfile.ZipFile(dml_nupkg_zip).namelist()
+
+            # ORT accepts win-x64 as platform, whereas DirectML accepts x64-win
+            platform_split = platform.strip().split("-")
+            platform = platform_split[1] + "-" + platform_split[0]
+
+            folder = "bin/" + platform
+            header_folder = "include"
+
+            check_if_headers_are_present(dml_related_header_files, header_folder, dml_zip_file_list, platform)
+            for dll in dml_related_dlls:
```

Comment:
Rename `dll` to `filename` or something more generic, as .lib and .pdb are not DLL's.

Reply:
Similarly `dml_related_dlls` should be `dml_related_filenames` because only 2 of them are DLL's.